london-class 8/responsive-grid-project/hamra-khan

HamraKhan opened the pull request.

HamraKhan committed:
Added base html for css grid

Added main sections for base of css grid including images

HamraKhan committed:
Added base for css grid layout

HamraKhan committed:
Created folder for images

HamraKhan committed:
Created icon folder

Added search icon and basket icon

HamraKhan committed:
Added html for top half of header section

- Referencing external font styles
- Added more html code to top half of header

HamraKhan committed:
Added css styling for top half of header

- Created and added color variables
- Inserted font families
-  Added css styling for search bar and basket icon

HamraKhan committed:
- Added logo and hamburger icon
-Added html for hero section
-Added html for shop section

HamraKhan committed:
- Added css for logo/navbar section
-Added css for hero section
-Added css for shop section

HamraKhan committed:
-  LInked header, about and form css files
-  Added classes for <hr> tabs
- Changed class names for button tabs
-  Added html for about section
- Added html for form section

HamraKhan committed:
- Changed grid template rows to auto value
-  Deleted and moved header to new header.css file
-  Removed padding bottom in hero class
-  Added class name to hr tab
- Changed button class name
-  Removed padding and margin to shop section

HamraKhan committed:
Added css styling to header css file

HamraKhan committed:
Added css styling for about us section to
about-us css file

HamraKhan committed:
Removed unecessary spacing

HamraKhan committed:
Added logo svg icon for hero section

HamraKhan committed:
Added html markup for form section

HamraKhan committed:
Added cursor pointer to button class

HamraKhan committed:
Added css styling for form section

HamraKhan committed:
- Added social media svg icons into icons folder

HamraKhan committed:
- Linked css footer file to html
-  Added html for end page image and footer, copyright sections

HamraKhan committed:
- Added copyright and credit to grid
template areas
-  changed max width from vw to %
-  added css styling for end page img and footer sections

HamraKhan committed:
removed comment

HamraKhan committed:
Added css styling for footer section

HamraKhan committed:
Added dropdown arrow svg

HamraKhan committed:
Added html markup for FAQ page

HamraKhan committed:
Added css styling to faq css file

HamraKhan committed:
Made changes to main css in copyright class and
title underline class

HamraKhan committed:
removed classes not being used from header
css file

HamraKhan committed:
Altered margin in title underline class and
added color to paragraph class

HamraKhan committed:
- Changed margin values
- Removed extra space
- Renamed class name
- Removed comments

HamraKhan committed:
Added svg icon for profile login

HamraKhan committed:
- Altered css grid
- Added media query

HamraKhan committed:
- Added content for account and nav menu bar
for desktop version
- Added class in shop section

HamraKhan committed:
-Added class names
-Altered class names
-Added media query

ewintram commented on css/faq.css:
```diff
@@ -0,0 +1,144 @@
+:root {
+    --red-brown: #A05941;
```
awesome to see your use of CSS variables! I think you only need to define them in the one CSS file that you import into all pages, i.e. main.css

ewintram commented on css/faq.css:
```diff
@@ -0,0 +1,144 @@
+    margin: 0;
+    grid-template-columns: repeat(4 1fr);
+    grid-template-rows: auto;
+    grid-template-areas: 
```
great use of all these CSS grid properties! I think it would make sense to define styles on the `body` in a CSS file that is shared across all pages (rather than in this page-specific file), as they should probably be consistent across the site.

ewintram commented on css/form.css:
```diff
@@ -0,0 +1,59 @@
+
+.form input[type=text], 
+     [type=email], 
+     [type=tel], 
```
well done targeting the different input types

ewintram commented on css/header.css:
```diff
@@ -0,0 +1,258 @@
+.header-nav {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
```
great use of flex-box

ewintram commented on css/header.css:
```diff
@@ -0,0 +1,258 @@
+}
+
+.nav-icon:hover:after {
+    transform: translateY(-12px) rotate(-135deg);
```
wow, well done achieving this transform effect!

ewintram commented on css/header.css:
```diff
@@ -0,0 +1,258 @@
+/* MEDIA QUERIES */
+
+/* tablet */
+@media screen and (min-width: 900px) {
```
good job building this mobile-first. It makes sense to keep media query styles close to the base styles they're overriding, so the `body` styles here might be better placed in main.css

ewintram commented on css/main.css:
```diff
@@ -1,3 +1,309 @@
+    flex-direction: column;
+    align-items: center;
+    background-size: cover;
+    background-repeat: no-repeat;
```
perfect use of a background image

ewintram commented on faq.html:
```diff
@@ -0,0 +1,216 @@
+        <title>Responsive grid project</title>
+        <link rel="stylesheet" href="./css/header.css"/>
+        <link rel="stylesheet" href="./css/faq.css"/>
+        <link rel="stylesheet" href="./css/footer.css"/>
```
I really like how you break down the CSS files as you are starting to think in terms of components

ewintram commented on faq.html:
```diff
@@ -0,0 +1,216 @@
+            <div class="header-top-half">
+                <!-- SEARCH BAR -->
+                <div class="search-container">
+                    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet"/>
```
I think `link` tags should be in the `head` element

ewintram commented on faq.html:
```diff
@@ -0,0 +1,216 @@
+                <!-- SEARCH BAR -->
+                <div class="search-container">
+                    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet"/>
+                    <i class="fa fa-search" aria-hidden="true" ></i>
```
nice us of FA icons!

ewintram commented on faq.html:
```diff
@@ -0,0 +1,216 @@
+        <header class="header">
+            <div class="header-top-half">
+                <!-- SEARCH BAR -->
+                <div class="search-container">
```
This div could actually be a form. There are some other attributes you can use to ensure an input of type `search` is used in an accessible way, check out the docs https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search

ewintram commented on faq.html:
```diff
@@ -0,0 +1,216 @@
+
+                <!-- SHOPPING BASKET -->
+                <div class="shopping-basket">
+                    <img src="./icons/shopping-basket.svg" class="svg-color" alt="shopping basket icon"/>
```
great that you're using svgs for icons

ewintram commented on faq.html:
```diff
@@ -0,0 +1,216 @@
+            
+            <!-- MAP DISPLAY -->
+            <div class="map-container">
+                <iframe 
```
nice job adding the map

ewintram reviewed (commented):
The site looks super! Well done for tackling 2 pages and still making your code really DRY

HamraKhan committed:
- Changed button and div class names
- Removed extra spacing

HamraKhan committed:
- Moved credit section into footer.css

HamraKhan committed:
- changed button class name in about-us.css
- added css styling for button and added styling for copyright section

HamraKhan committed:
- Added class name to h1 and h2 tags
-  added class name for arrown svg
-  Added styling for faq.css page

HamraKhan committed:
- Removed code repetition
-  Added media queries

HamraKhan committed:
- Added shared.css and moved link to head
- Changed div to form tab
- Added div conatainer and wrapper to about us section

HamraKhan committed:
- Added shared.css to head
- Added html menu bar for desktop version

HamraKhan committed:
- Removed media queries
- Changed css styles
-  Added color variables to shared css file
-  Removed root color variables

HamraKhan committed:
- Moved end page image into form section
- Changed div to form tag
- Added wrapper for media queries and removed class names in img tag

HamraKhan committed:
- Added more css styling for about-us
-  Adjusted grid template areas in faq.css and added media queries
for page

HamraKhan committed:
- Added media queries for footer.css
- Added media queries for form.css

HamraKhan committed:
- Added media queries for header.css
- Added color varible to shared root

HamraKhan committed:
- Removed media queries for header section
